packages/mds-ingest-service/@types/index.ts

```diff
@@ -0,0 +1,67 @@
+import { DomainModelCreate } from '@mds-core/mds-repository'
+import {
+  Nullable,
+  NullableOptional,
+  PROPULSION_TYPE,
+  TelemetryData,
+  Timestamp,
+  UUID,
+  VEHICLE_EVENT,
+  VEHICLE_REASON,
+  VEHICLE_TYPE
+} from '@mds-core/mds-types'
+import { RpcServiceDefinition, RpcRoute } from '@mds-core/mds-rpc-common'
+
+export interface DeviceDomainModel {
+  device_id: UUID
+  provider_id: UUID
+  vehicle_id: string
+  type: VEHICLE_TYPE
+  propulsion: PROPULSION_TYPE[]
+  recorded: Timestamp
+
+  year: Nullable<number>
+  mfgr: Nullable<string>
+  model: Nullable<string>
+}
+
+export type DeviceDomainCreateModel = DomainModelCreate<DeviceDomainModel>
+
+/* More flexible version of WithGpsProperty */
+type WithGpsPropertyFlex<T extends TelemetryData> = Omit<T, keyof Omit<TelemetryData, 'charge'>> & {
+  gps: Omit<T, 'charge'>
+}
```

Comment:
You could avoid the `Flex` suffix and make this even more flexible with something like:
```ts
type WithGpsData<T extends TelemetryData, P extends string = 'gps'> = Omit<T, keyof Omit<TelemetryData, 'charge'>> & {
  P: Omit<T, 'charge'>
}
```